src/modules/GroupChannel/context/GroupChannelProvider.tsx

```diff
@@ -198,117 +196,147 @@ const GroupChannelProvider = (props: GroupChannelContextProps) => {
     replyType: chatReplyType,
     startingPoint,
+    shouldCountNewMessages: () => !isScrollBottomReached,
     markAsRead: (channels) => {
       if (!disableMarkAsRead && isScrollBottomReached) {
         channels.forEach((it) => markAsReadScheduler.push(it));
       }
     },
-    shouldCountNewMessages: () => !isScrollBottomReached,
     onMessagesReceived: () => {
       if (isScrollBottomReached && isContextMenuClosed()) {
-        scrollToBottom();
+        scrollPubSub.publish('scrollToBottom', null);
       }
     },
-    // collectionCreator?: (collectionParams?: DefaultCollectionParams) => MessageCollection;
-    // onMessagesUpdated?: (messages: SendbirdMessage[]) => void;
-    // onChannelDeleted?: (channelUrl: string) => void;
-    // onChannelUpdated?: (channel: GroupChannel) => void;
-    // onCurrentUserBanned?: () => void;
+    onChannelDeleted: () => setCurrentChannel(null),
+    onCurrentUserBanned: () => setCurrentChannel(null),
+    onChannelUpdated: (channel) => setCurrentChannel(channel),
     logger,
+    // collectionCreator?: (collectionParams?: DefaultCollectionParams) => MessageCollection;
   });
 
+  const preventDuplicateRequest = usePreventDuplicateRequest();
   useOnScrollPositionChangeDetectorWithRef(scrollRef, {
-    onReachedTop({ distanceFromBottom }) {
-      setIsScrollBottomReached(false);
-      scrollDistanceFromBottomRef.current = distanceFromBottom;
-
-      // TODO: maintain the view position.
-      const prevViewInfo = {
-        scrollTop: scrollRef.current.scrollTop,
-        scrollHeight: scrollRef.current.scrollHeight,
-        clientHeight: scrollRef.current.clientHeight,
-      };
-      messageCollectionHook.loadPrevious().then(() => {
+    async onReachedTop() {
+      preventDuplicateRequest.lock();
+
+      await preventDuplicateRequest.run(async () => {
+        if (!messageDataSource.hasPrevious()) return;
+
+        const prevViewInfo = { scrollTop: scrollRef.current.scrollTop, scrollHeight: scrollRef.current.scrollHeight };
+        await messageDataSource.loadPrevious();
+
         setTimeout(() => {
-          const nextViewInfo = {
-            scrollTop: scrollRef.current.scrollTop,
-            scrollHeight: scrollRef.current.scrollHeight,
-            clientHeight: scrollRef.current.clientHeight,
-          };
-
-          if (prevViewInfo.scrollTop === 0 && nextViewInfo.scrollTop) {
-            const topOffset = nextViewInfo.scrollHeight - prevViewInfo.scrollHeight;
-            if (topOffset > 0) scrollRef.current.scrollTop = topOffset;
+          const nextViewInfo = { scrollTop: scrollRef.current.scrollTop, scrollHeight: scrollRef.current.scrollHeight };
+          const viewUpdated = prevViewInfo.scrollHeight < nextViewInfo.scrollHeight;
+
+          if (viewUpdated) {
+            const bottomOffset = prevViewInfo.scrollHeight - prevViewInfo.scrollTop;
+            scrollPubSub.publish('scroll', { top: nextViewInfo.scrollHeight - bottomOffset, lazy: false });
```

Comment:
seems only `nextViewInfo.scrollHeight` is being used, do we need to put the other properties?